File: src/Entity/Paiement.php
<?php

namespace App\Entity;

use App\Repository\PaiementRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: PaiementRepository::class)]
#[ORM\Table(name: 'paiement')]
class Paiement
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(type: 'integer', nullable: true)]
    private ?int $commandeId = null;

    #[ORM\Column(type: 'integer', nullable: true)]
    private ?int $demandeId = null;

    #[ORM\Column(type: 'integer', nullable: true)]
    private ?int $clientId = null;

    #[ORM\Column(type: 'integer', nullable: true)]
    private ?int $activiteId = null;

    #[ORM\Column(type: 'decimal', precision: 10, scale: 2)]
    private float $montant = 0.0;

    #[ORM\Column(length: 100)]
    private string $methodePaiement = '';

    #[ORM\Column(nullable: true, length: 50)]
    private ?string $statut = null;

    #[ORM\Column(type: 'datetime', nullable: true)]
    private ?\DateTimeInterface $datePaiement = null;

    #[ORM\Column(nullable: true, length: 255)]
    private ?string $transactionId = null;

    #[ORM\Column(nullable: true, length: 255)]
    private ?string $referenceExterne = null;

    #[ORM\Column(nullable: true, length: 255)]
    private ?string $referenceTransaction = null;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $detailsJson = null;

    #[ORM\Column(type: 'datetime', nullable: true)]
    private ?\DateTimeInterface $dateCreation = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getCommandeId(): ?int
    {
        return $this->commandeId;
    }

    public function setCommandeId(?int $commandeId): static
    {
        $this->commandeId = $commandeId;
        return $this;
    }

    public function getDemandeId(): ?int
    {
        return $this->demandeId;
    }

    public function setDemandeId(?int $demandeId): static
    {
        $this->demandeId = $demandeId;
        return $this;
    }

    public function getClientId(): ?int
    {
        return $this->clientId;
    }

    public function setClientId(?int $clientId): static
    {
        $this->clientId = $clientId;
        return $this;
    }

    public function getActiviteId(): ?int
    {
        return $this->activiteId;
    }

    public function setActiviteId(?int $activiteId): static
    {
        $this->activiteId = $activiteId;
        return $this;
    }

    public function getMontant(): float
    {
        return $this->montant;
    }

    public function setMontant(float $montant): static
    {
        $this->montant = $montant;
        return $this;
    }

    public function getMethodePaiement(): string
    {
        return $this->methodePaiement;
    }

    public function setMethodePaiement(string $methodePaiement): static
    {
        $this->methodePaiement = $methodePaiement;
        return $this;
    }

    public function getStatut(): ?string
    {
        return $this->statut;
    }

    public function setStatut(?string $statut): static
    {
        $this->statut = $statut;
        return $this;
    }

    public function getDatePaiement(): ?\DateTimeInterface
    {
        return $this->datePaiement;
    }

    public function setDatePaiement(?\DateTimeInterface $datePaiement): static
    {
        $this->datePaiement = $datePaiement;
        return $this;
    }

    public function getTransactionId(): ?string
    {
        return $this->transactionId;
    }

    public function setTransactionId(?string $transactionId): static
    {
        $this->transactionId = $transactionId;
        return $this;
    }

    public function getReferenceExterne(): ?string
    {
        return $this->referenceExterne;
    }

    public function setReferenceExterne(?string $referenceExterne): static
    {
        $this->referenceExterne = $referenceExterne;
        return $this;
    }

    public function getReferenceTransaction(): ?string
    {
        return $this->referenceTransaction;
    }

    public function setReferenceTransaction(?string $referenceTransaction): static
    {
        $this->referenceTransaction = $referenceTransaction;
        return $this;
    }

    public function getDetailsJson(): ?string
    {
        return $this->detailsJson;
    }

    public function setDetailsJson(?string $detailsJson): static
    {
        $this->detailsJson = $detailsJson;
        return $this;
    }

    public function getDateCreation(): ?\DateTimeInterface
    {
        return $this->dateCreation;
    }

    public function setDateCreation(?\DateTimeInterface $dateCreation): static
    {
        $this->dateCreation = $dateCreation;
        return $this;
    }

}

File: nexora_java/src/main/java/com/pi/entities/Paiement.java
package com.pi.entities;

import java.time.LocalDateTime;

public class Paiement {
    private int id;
    private int demandeId;
    private int clientId;
    private int activiteId;
    private double montant;
    private String methodePaiement;
    private String statut; // EN_COURS, COMPLETE, ECHOUE
    private LocalDateTime datePaiement;
    private String referenceTransaction;

    public Paiement() {
        this.datePaiement = LocalDateTime.now();
        this.statut = "EN_COURS";
    }

    public Paiement(int demandeId, int clientId, int activiteId, double montant, String methodePaiement) {
        this.demandeId = demandeId;
        this.clientId = clientId;
        this.activiteId = activiteId;
        this.montant = montant;
        this.methodePaiement = methodePaiement;
        this.datePaiement = LocalDateTime.now();
        this.statut = "EN_COURS";
    }

    // Getters et Setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getDemandeId() {
        return demandeId;
    }

    public void setDemandeId(int demandeId) {
        this.demandeId = demandeId;
    }

    public int getClientId() {
        return clientId;
    }

    public void setClientId(int clientId) {
        this.clientId = clientId;
    }

    public int getActiviteId() {
        return activiteId;
    }

    public void setActiviteId(int activiteId) {
        this.activiteId = activiteId;
    }

    public double getMontant() {
        return montant;
    }

    public void setMontant(double montant) {
        this.montant = montant;
    }

    public String getMethodePaiement() {
        return methodePaiement;
    }

    public void setMethodePaiement(String methodePaiement) {
        this.methodePaiement = methodePaiement;
    }

    public String getStatut() {
        return statut;
    }

    public void setStatut(String statut) {
        this.statut = statut;
    }

    public LocalDateTime getDatePaiement() {
        return datePaiement;
    }

    public void setDatePaiement(LocalDateTime datePaiement) {
        this.datePaiement = datePaiement;
    }

    public String getReferenceTransaction() {
        return referenceTransaction;
    }

    public void setReferenceTransaction(String referenceTransaction) {
        this.referenceTransaction = referenceTransaction;
    }

    @Override
    public String toString() {
        return "Paiement{" +
                "id=" + id +
                ", montant=" + montant +
                ", statut='" + statut + '\'' +
                ", datePaiement=" + datePaiement +
                '}';
    }
}
